### PR TITLE
Benchmark: Fixes to show estimated cost of a container only when new container is getting created

### PR DESCRIPTION
**Pull request by sourabh1007**
## Description

Every time when benchmark tests are running, it is showing container estimated cost and waiting for the user input to continue. Due to which benchmarks were not running on the VM.
As part of this PR, 
1. Removing estimated cost message.
2. disabling telemetry feature by default in config, let the script decide if telemetry should be enabled or not.
## Type of change
- [] Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

**Review by philipthomas-MSFT — APPROVED** [review]

File: Microsoft.Azure.Cosmos.Samples/Tools/Benchmark/BenchmarkConfig.cs [hunk 1]
```diff
@@ -103,8 +103,8 @@ public class BenchmarkConfig
         [Option(Required = false, HelpText = "Disable core SDK logging")]
         public bool DisableCoreSdkLogging { get; set; }
 
-        [Option(Required = false, HelpText = "Disable Distributed Tracing feature from source")]
-        public bool DisableDistributedTracing { get; set; } = false;
```
Comment (ealsur) [inline Microsoft.Azure.Cosmos.Samples/Tools/Benchmark/BenchmarkConfig.cs:107]: How are these changes related to what is explained in the Description?

File: Microsoft.Azure.Cosmos.Samples/Tools/Benchmark/Program.cs [hunk 2]
```diff
@@ -320,19 +317,29 @@ private static Type[] AvailableBenchmarks()
...
+                container = database.GetContainer(options.Container);
+            }
+            catch (CosmosException ex ) when (ex.StatusCode == HttpStatusCode.NotFound)
+            {
+                // Show user cost of running this test only when it is creating new container
+                double estimatedCostPerMonth = 0.06 * options.Throughput;
+                double estimatedCostPerHour = estimatedCostPerMonth / (24 * 30);
+                Utility.TeeTraceInformation($"The container {options.Container} will cost an estimated ${Math.Round(estimatedCostPerHour, 2)} per hour (${Math.Round(estimatedCostPerMonth, 2)} per month)");
+                Utility.TeeTraceInformation("Press enter to continue ...");
+                Console.ReadLine();
```
Comment (ealsur) [inline Microsoft.Azure.Cosmos.Samples/Tools/Benchmark/Program.cs:336]: @kundadebdatta Do you recall if we need this ReadLine here? If customers are running this as an ARM Template, is pressing ENTER something that they can do? Should we simply remove the ReadLine

File: Microsoft.Azure.Cosmos.Samples/Tools/Benchmark/Program.cs [hunk 3]
```diff
@@ -320,19 +317,29 @@ private static Type[] AvailableBenchmarks()
...
-            // Show user cost of running this test
-            double estimatedCostPerMonth = 0.06 * options.Throughput;
-            double estimatedCostPerHour = estimatedCostPerMonth / (24 * 30);
-            Utility.TeeTraceInformation($"The container will cost an estimated ${Math.Round(estimatedCostPerHour, 2)} per hour (${Math.Round(estimatedCostPerMonth, 2)} per month)");
-            Utility.TeeTraceInformation("Press enter to continue ...");
-            Console.ReadLine();
+            Container container;
+            try
+            {
+                container = database.GetContainer(options.Container);
```
Comment (ealsur) [inline Microsoft.Azure.Cosmos.Samples/Tools/Benchmark/Program.cs:327]: This is wrong, GetContainer does not perform any network operation, this cannot be throwing any exception

**Review by ealsur — CHANGES_REQUESTED** [review]

File: Microsoft.Azure.Cosmos.Samples/Tools/Benchmark/Program.cs (hunk 3 above)
Comment (ealsur) [reply Microsoft.Azure.Cosmos.Samples/Tools/Benchmark/Program.cs:327]: And it is not checking if the container exists.

File: Microsoft.Azure.Cosmos.Samples/Tools/Benchmark/BenchmarkConfig.cs (hunk 1 above)
Comment (sourabh1007) [reply Microsoft.Azure.Cosmos.Samples/Tools/Benchmark/BenchmarkConfig.cs:107]: 2nd point above is related to this, I renamed the config name because it sounds better than having config name with prefix disabled.

File: Microsoft.Azure.Cosmos.Samples/Tools/Benchmark/Program.cs [hunk 4]
```diff
@@ -320,19 +317,29 @@ private static Type[] AvailableBenchmarks()
         /// Get or Create a partitioned container and display cost of running this test.
         /// </summary>
         /// <returns>The created container.</returns>
-        private static async Task<ContainerResponse> CreatePartitionedContainerAsync(BenchmarkConfig options, CosmosClient cosmosClient)
+        private static async Task<Container> CreatePartitionedContainerAsync(BenchmarkConfig options, CosmosClient cosmosClient)
         {
             Microsoft.Azure.Cosmos.Database database = await cosmosClient.CreateDatabaseIfNotExistsAsync(options.Database);
 
-            // Show user cost of running this test
-            double estimatedCostPerMonth = 0.06 * options.Throughput;
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos.Samples/Tools/Benchmark/Program.cs:328]: How relevant is this message? 
Am for getting rid of it.

File: Microsoft.Azure.Cosmos.Samples/Tools/Benchmark/Program.cs (hunk 3 above)
Comment (sourabh1007) [reply Microsoft.Azure.Cosmos.Samples/Tools/Benchmark/Program.cs]: My bad....i misunderstood it while taking it from history. fixed it.

File: Microsoft.Azure.Cosmos.Samples/Tools/Benchmark/Program.cs (hunk 4 above)
Comment (sourabh1007) [reply Microsoft.Azure.Cosmos.Samples/Tools/Benchmark/Program.cs:328]: it is not relevant for us because internally we really don't care. Might be useful for external customers.

File: Microsoft.Azure.Cosmos.Samples/Tools/Benchmark/Program.cs (hunk 2 above)
Comment (kundadebdatta) [reply Microsoft.Azure.Cosmos.Samples/Tools/Benchmark/Program.cs:338]: I agree we should remove the `ReadLine()`. However, regarding all the benchmarking demo we had, I have never seen David pressed an enter to progress the benchmark. I will double check how this may impact the existing benchmark, but it it did, our benchmarking tests would hanged for indefinite period.

Comment (kundadebdatta) [reply Microsoft.Azure.Cosmos.Samples/Tools/Benchmark/Program.cs]: Oh wait - It's applicable for new containers. Hmm, thinking more on this, this should be removed.

File: Microsoft.Azure.Cosmos.Samples/Tools/Benchmark/Program.cs (hunk 4 above)
Comment (sourabh1007) [reply Microsoft.Azure.Cosmos.Samples/Tools/Benchmark/Program.cs:328]: offline synced with @kundadebdatta, External customers don't see this message. Hence removing it.

File: Microsoft.Azure.Cosmos.Samples/Tools/Benchmark/Program.cs (hunk 2 above)
Comment (kundadebdatta) [reply Microsoft.Azure.Cosmos.Samples/Tools/Benchmark/Program.cs]: And, this is actually inherited from the earlier version of the code. See [these lines of code in SDK release 3.29.0](https://github.com/Azure/azure-cosmos-dotnet-v3/blob/releases/3.29.0/Microsoft.Azure.Cosmos.Samples/Tools/Benchmark/Program.cs#L275). So I think now is a good time to remove these codes. Cc: @ealsur , @sourabh1007 , @kirankumarkolli .

**Review by kundadebdatta — DISMISSED** [review]
LGTM.

File: Microsoft.Azure.Cosmos.Samples/Tools/Benchmark/Program.cs [hunk 5]
```diff
@@ -323,16 +321,9 @@ private static Type[] AvailableBenchmarks()
...
-            double estimatedCostPerHour = estimatedCostPerMonth / (24 * 30);
-            Utility.TeeTraceInformation($"The container will cost an estimated ${Math.Round(estimatedCostPerHour, 2)} per hour (${Math.Round(estimatedCostPerMonth, 2)} per month)");
-            Utility.TeeTraceInformation("Press enter to continue ...");
-            Console.ReadLine();
-
+            
             string partitionKeyPath = options.PartitionKeyPath;
-            return await database.CreateContainerIfNotExistsAsync(options.Container, partitionKeyPath, options.Throughput);
+            return  await database.CreateContainerIfNotExistsAsync(options.Container, partitionKeyPath, options.Throughput);
```
Comment (ealsur) [inline Microsoft.Azure.Cosmos.Samples/Tools/Benchmark/Program.cs:326]:
```suggestion
            return await database.CreateContainerIfNotExistsAsync(options.Container, partitionKeyPath, options.Throughput);
```

File: Microsoft.Azure.Cosmos.Samples/Tools/Benchmark/Program.cs [hunk 6]
```diff
@@ -156,8 +154,8 @@ private async Task<RunSummary> ExecuteAsync(BenchmarkConfig config)
                 }
 
                 ContainerResponse containerResponse = await Program.CreatePartitionedContainerAsync(config, cosmosClient);
-                Container container = containerResponse;
 
+                Container container = containerResponse;
```
Comment (ealsur) [inline Microsoft.Azure.Cosmos.Samples/Tools/Benchmark/Program.cs:158]: undo this change?

**Review by ealsur — APPROVED** [review]

File: Microsoft.Azure.Cosmos.Samples/Tools/Benchmark/Program.cs (hunk 6 above)
Comment (ealsur) [reply Microsoft.Azure.Cosmos.Samples/Tools/Benchmark/Program.cs:158]: Meaning this change just moved the code 2 lines lower than it was

**Review by kirankumarkolli — DISMISSED** [review]

File: Microsoft.Azure.Cosmos.Samples/Tools/Benchmark/Program.cs (hunk 6 above)
Comment (sourabh1007) [reply Microsoft.Azure.Cosmos.Samples/Tools/Benchmark/Program.cs]: I don't know, how does it matter. but anyway reverted it.

**Review by ealsur — APPROVED** [review]

File: Microsoft.Azure.Cosmos.Samples/Tools/Benchmark/BenchmarkConfig.cs [hunk 7]
```diff
@@ -103,8 +103,8 @@ public class BenchmarkConfig
...
+        [Option(Required = false, HelpText = "Enable Distributed Tracing")]
+        public bool EnableDistributedTracing { get; set; } = false;
```
Comment (ealsur) [inline Microsoft.Azure.Cosmos.Samples/Tools/Benchmark/BenchmarkConfig.cs:107]: nit: Are there any script files that need updating in the repo? Like the ones that launch the benchmark process?

**Review by kirankumarkolli — APPROVED** [review]

File: Microsoft.Azure.Cosmos.Samples/Tools/Benchmark/BenchmarkConfig.cs (hunk 7 above)
Comment (sourabh1007) [reply Microsoft.Azure.Cosmos.Samples/Tools/Benchmark/BenchmarkConfig.cs:107]: scripts are already using this new name. 

I changed these config name in public API PR where I didn't change the scripts. and reverted these names in this PR as it was not looking script friendly names.
for ref. https://github.com/Azure/azure-cosmos-dotnet-v3/blob/7ed22ac55b569354ea94733ea5ab4601d20dd586/Microsoft.Azure.Cosmos.Samples/Tools/Benchmark/run.sh#L54
